core/types/access_list_tx.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2020 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package types
+
+import (
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+type AccessTuple struct {
+	Address     *common.Address `json:"address"        gencodec:"required"`
+	StorageKeys []*common.Hash  `json:"storageKeys"    gencodec:"required"`
+}
```

Comment:
I don't really see a reason why these would be pointers internally. Unless I'm missing something, we should probably fix this.

Reply:
Also please add type and method docs throughout

Reply:
Fixed